libraries/AP_BattMonitor/AP_BattMonitor.cpp

```diff
@@ -254,9 +282,8 @@ float AP_BattMonitor::voltage(uint8_t instance) const
 /// this will always be greater than or equal to the raw voltage
 float AP_BattMonitor::voltage_resting_estimate(uint8_t instance) const
 {
-    if (instance < _num_instances) {
-        // resting voltage should always be greater than or equal to the raw voltage
-        return MAX(state[instance].voltage, state[instance].voltage_resting_estimate);
+    if (instance < _num_instances && drivers[instance] != nullptr) {
```

Comment:
Can `_num_instances` ever drop?  i.e. is this over-cautious now?

Reply:
No, but there are a lot of issues associated with `_num_instances`. It should never have been used, but it's a large (and very different patch set) to get it backed out. If a user ever sets up just the second batt monitor bad things happen.